shogun-lib/src/main/java/de/terrestris/shogun/lib/service/ImageFileService.java

```diff
@@ -89,4 +95,60 @@ public void isValidType(String contentType) throws InvalidContentTypeException {
         }
     }
 
+    @Override
+    public ImageFile create(MultipartFile uploadFile, Boolean writeToSystem) throws Exception {
+        if (!writeToSystem) {
+            return this.create(uploadFile);
+        }
+
+        String uploadBasePath = uploadProperties.getPath();
+        String filename = uploadFile.getOriginalFilename();
+
+        FileUtil.validateFile(uploadFile);
+
+        byte[] fileByteArray = FileUtil.fileToByteArray(uploadFile);
+
+        ImageFile file = new ImageFile();
+        file.setFileType(uploadFile.getContentType());
+        file.setFileName(uploadFile.getOriginalFilename());
+        file.setActive(true);
+
+        Dimension imageDimensions = ImageFileUtil.getImageDimensions(uploadFile);
+        if (imageDimensions != null) {
+            int thumbnailSize = uploadProperties.getImage().getThumbnailSize();
+            file.setThumbnail(ImageFileUtil.getScaledImage(uploadFile, imageDimensions, thumbnailSize));
+            file.setWidth(imageDimensions.width);
+            file.setHeight(imageDimensions.height);
+        } else {
+            LOG.warn("Could not detect the dimensions of the image. Neither width, height " +
+                "nor the thumbnail can be set.");
+        }
+
+        ImageFile savedFile = this.create(file);
+        UUID fileUuid = savedFile.getFileUuid();
+
+        // Setup path and directory
+        String path = fileUuid + "/" + filename;
+        java.io.File fileDirectory = new java.io.File(uploadBasePath + "/" + fileUuid);
+        fileDirectory.mkdirs();
+
+        // Write multipart file data to target directory
+        java.io.File outFile = new java.io.File(fileDirectory, filename);
+        InputStream in = new ByteArrayInputStream(fileByteArray);
+
+        try (OutputStream out = new FileOutputStream(outFile)) {
+            IOUtils.copy(in, out);
+            LOG.info("Saved file with id {} to {}: ", savedFile.getId(), savedFile.getPath());
+        } catch (Exception e) {
+            LOG.error("Error when saving file {} to disk: " + e.getMessage(), savedFile.getId());
+            throw e;
+        }
+
+        // Update entity with saved File
+        savedFile.setPath(path);
+        this.repository.save(savedFile);
```

Comment:
I think it's better to return the output of `this.repository.save(savedFile)` instead of `return savedFile;`